### PR TITLE
feat(ui): add DHCP snippets table to subnet and vlan details

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Update the DHCP snippet table to handle subnets.
- Add the DHCP snippets table to subnet and vlan details.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to Settings -> DHCP snippets.
- Add a DHCP snippet to a subnet.
- Go to the React subnet details for that subnet. You should see a table with the DHCP snippet you added.
- Go to the subnet's VLAN details. You should see a table with the DHCP snippet you added.

## Fixes

Fixes: canonical-web-and-design/app-tribe#672.
Fixes: canonical-web-and-design/app-tribe#716.
Fixes: canonical-web-and-design/app-tribe#673.

### CODE REVIEW

File: ui/src/app/subnets/components/DHCPSnippets/DHCPSnippets.test.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  expectedActions.forEach((expectedAction) => {
+    expect(
+      actualActions.find(
+        (actualAction) => actualAction.type === expectedAction.type
+      )
+    ).toStrictEqual(expectedAction);
+  });
+});
+
+it("displays a message if the subnet does not exist", () => {
```
Comment (Caleb-Ellis) [inline ui/src/app/subnets/components/DHCPSnippets/DHCPSnippets.test.tsx:52]: This isn't right

File: ui/src/app/subnets/views/VLANDetails/VLANDetails.tsx [hunk 2]
```diff
@@ -77,7 +84,10 @@ const VLANDetails = (): JSX.Element => {
       <DHCPStatus />
       <ReservedRanges />
       <VLANSubnets />
-      <DHCPSnippets />
+      <DHCPSnippets
+        subnetIds={subnets.map(({ id }) => id)}
```
Comment (Caleb-Ellis) [inline ui/src/app/subnets/views/VLANDetails/VLANDetails.tsx:88]: There's no list of subnet IDs on the vlan object? That's lame. Maybe we should request an update to the websocket API?

**Review by Caleb-Ellis — APPROVED** [review]
LGTM just a few test descriptions to update

File: ui/src/app/subnets/views/VLANDetails/VLANDetails.tsx (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/subnets/views/VLANDetails/VLANDetails.tsx:88]: Good idea, I filed: https://bugs.launchpad.net/maas/+bug/1959648.

**Pull request merged**